pkg/machines/components/diskEdit.jsx

```diff
@@ -0,0 +1,213 @@
+                       checked={dialogValues.shareable}
+                       onChange={e => {
+                           onValueChanged('readonly', false);
+                           onValueChanged('shareable', true);
+                       }}
+                       className={dialogValues.shareable ? "active" : ''} />
+                {_("Writeable and shared")}
+            </label>}
+        </>
+    );
+};
+
+class EditDiskModalBody extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            readonly: props.disk.readonly,
+            shareable: props.disk.shareable,
+        };
+        this.onValueChanged = this.onValueChanged.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+        this.onSaveClicked = this.onSaveClicked.bind(this);
+    }
+
+    onValueChanged(key, value) {
+        this.setState({ [key]: value });
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    onSaveClicked() {
+        const { disk, vm } = this.props;
+        changeDiskSettings(vm.connectionName, vm.id, disk.target, this.state.readonly, this.state.shareable)
+                .then(() => this.props.close())
+                .fail((exc) => {
+                    this.dialogErrorSet(_("Disk settings could not be saved"), exc.message);
+                });
+    }
+
+    render() {
+        const { vm, disk } = this.props;
+        const idPrefix = `${this.props.idPrefix}-edit`;
+
+        const defaultBody = (
+            <div className='ct-form'>
+                <NameRow idPrefix={idPrefix}
+                         diskType={vm.disks[disk.target].type}
+                         name={getDiskFullName(vm.disks[disk.target])} />
+
+                <AccessRow dialogValues={this.state}
+                           idPrefix={idPrefix}
+                           driverType={vm.disks[disk.target].driver.type}
+                           onValueChanged={this.onValueChanged} />
+            </div>
+        );
+
+        const showFooterWarning = () => {
+            if (vm.state === 'running' && (
+                this.state.readonly !== disk.readonly ||
+                this.state.shareable !== disk.shareable)) {
+                return (
+                    <span id={`${idPrefix}-idle-message`} className='idle-message'>
+                        <i className='pficon pficon-pending' />
+                        <span>{_("Changes will take effect after shutting down the VM")}</span>
+                    </span>
+                );
+            }
+        };
+
+        return (
+            <Modal id={`${idPrefix}-dialog`} show onHide={this.props.close}>
+                <Modal.Header>
+                    <Modal.CloseButton onClick={this.props.close} />
+                    <Modal.Title> {'Edit ' + getDiskPrettyName(vm.disks[disk.target]) + ' Attributes'} </Modal.Title>
```

Comment:
I don't like the Idea of having 'Attributes' in the title, but if it was recommended by the designers I withdraw :)